src/azure-cli/azure/cli/command_modules/profile/custom.py

```diff
@@ -145,7 +162,21 @@ def login(cmd, username=None, password=None, service_principal=None, tenant=None
         scopes=scopes,
         use_device_code=use_device_code,
         allow_no_subscriptions=allow_no_subscriptions,
-        use_cert_sn_issuer=use_cert_sn_issuer)
+        use_cert_sn_issuer=use_cert_sn_issuer,
+        show_progress=select_subscription)
+
+    # No JSON output if interactive account selection is used
+    if select_subscription:
+        from ._subscription_selector import SubscriptionSelector
+        from azure.cli.core._profile import _SUBSCRIPTION_ID
+
+        selected = SubscriptionSelector(subscriptions)()
```

Comment:
`subscriptions` here only contain subscriptions accessible by this user, provided to `az login`. Should we also show subscriptions from other users as candidates?

Reply:
If the user does not have access to the subscription, why should be show it?

Reply:
Even if the current user doesn't have access, another user used in a previous `az login` has. Running `az account set` will automatically switch to the other user.